walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,95 @@
 
 package com.google.sps;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
 import java.util.Collection;
+import java.util.Collections;
+
 
 public final class FindMeetingQuery {
+
+  /** 
+   * Return all possible time ranges to hold a meeting of specific duration.
+  */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // Find which times work for mandatory attendees
+    ImmutableList<TimeRange> sortedBusyRangesForMandatoryAttendees = findBusyRanges(request.getAttendees(), events);
+    ImmutableList<TimeRange> freeRangesForMandatoryAttendees = findFreeRanges(request, sortedBusyRangesForMandatoryAttendees);
```

Comment:
Since you may not even use this if you handle the cases with optional attendees, you can move this line to just before returning line 49 instead.